tests/spec_tests/query_tests/distinct.yml

```diff
@@ -0,0 +1,132 @@
+catalog_data:
+  db:
+    foo:
+      - { _id: 1, a: 1, b: 2 }
+      - { _id: 2, a: 1, b: 2 }
+      - { _id: 3, a: 2, b: 2 }
+      - { _id: 4, b: 2, a: 1 }
+      - { _id: 5, a: 2, b: 2, c: true, d: "a", e: null }
+      - { _id: 6, a: 2, b: 2, c: true, d: "a", e: null }
+
+    documents:
+      - { _id: 1, doc: { a: 1, b: 2 } }
+      - { _id: 2, doc: { a: 1, b: 2 } }
+      - { _id: 3, doc: { a: 2, b: 2 } }
+      - { _id: 4, doc: { a: 2, b: 3 } }
+      - { _id: 5, doc: { b: 2, a: 1 } }
+
+    arrays:
+      - { _id: 1, arr: [1, 2, 3] }
+      - { _id: 2, arr: [1, 2, 3] }
+      - { _id: 3, arr: [2, 3, 4] }
+      - { _id: 4, arr: [3, 2, 1] }
+
+catalog_schema:
+  {
+    "db": {
+      "foo": {
+        "bsonType": "object",
+        "required": ["_id", "a", "b"],
+        "additionalProperties": false,
+        "properties": {
+          "_id":   { "bsonType": "int" },
+          "a":     { "bsonType": "int" },
+          "b":     { "bsonType": "int" },
+          "c": {
+            "anyOf": [
+              { "bsonType": "bool" },
+              { "bsonType": "null" }
```

Comment:
I'm glad to see this works without the need for `!!str`. I thought that was still a requirement but it appears not!